### ISSUE TITLE
Non-rooted new patching method solution

### ISSUE THREAD

**Issue by michelerenzullo** [itsMoji/Instagram_SSL_Pinning#136]
Hi itsMoji,

I am Michele Renzullo and I had the same problem with Workplace app, basically to create a non-rooted version as you already know we have to manipulate libs.spo that is now encrypted or using a strange compression, I was studying libsuperpack-jni and readNative() with a decompiler but I ended up creating a Java/dalvik-bytecode function to patch the offset of the library and implementing it in one of any main activity(I used the login one). I did for Workplace app and it works perfectly, you can do also for Instagram, everything is very well explained in my article. I think it is useful for a lot of android apps!
 [https://renzullomichele.medium.com/patch-encrypted-libraries-in-metas-apps-for-android-ad3dfd35db79](Medium)

If you find it useful please put some claps, for any improvements or suggestions I'm open to.

**Comment by 350d** [comment]
Hi @michelerenzullo! Please fork this repo or create your own for Instagram app. Very promising technique you've shared here. I hope it will be not so hard and time consuming for you to support SLL unpinned Instagram app one a month or so. Thank you!

**Comment by michelerenzullo** [comment]
Yeah, I think I am gonna do, already forked this repo last year, in a few days I could publish a non-rooted version for Instagram as well.

**Comment by 350d** [comment]
@michelerenzullo Sounds great! Thanks!

**Comment by NickParks** [comment]
Hi Michel, would love to get an APK if available. Thank you!

**Comment by itsMoji** [comment]
Hi @michelerenzullo,
Thanks for sharing this info.

I've already been able to decompress and rebuild the libs.spo; the problem is not with this file.
The [libs](https://github.com/itsMoji/Instagram_SSL_Pinning/tree/master/dev/libs) file in the [dev/](https://github.com/itsMoji/Instagram_SSL_Pinning/tree/master/dev) folder was inside the libs.spo, and I don't think it's a superpack file.

**Comment by itsMoji** [comment]
It was just an obfuscation,
ARM patched APK updated to the latest version.

**Comment by michelerenzullo** [comment]
More info about?

**Comment by keralaboy123** [comment]
Hi have you built apk for instagram/workplace then please share .itsmojis app is not working

**Comment by michelerenzullo** [comment]
Which version of itsMoji is not working? I think has been patched correctly I checked the libs.xz archive and he replaced the 4 bytes and update the size of the file in the header
